src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-server.cpp

```diff
@@ -254,18 +256,22 @@ void WiFiDiagosticsGlobalInstance::InvokeCommand(HandlerContext & handlerContext
 {
     switch (handlerContext.mRequestPath.mCommandId)
     {
+#ifdef WI_FI_NETWORK_DIAGNOSTICS_ENABLE_RESET_COUNTS_CMD        
```

Comment:
If we only keep this, would link-time-optimization already remove out the flash? trying to figure out the minimal change required and this seems to be what ember generally does (just does not have calls inside its switch ... although on free-standing functions)

Reply:
No. The linker cannot automatically remove the code from flash if it is referenced in the switch statement. As long as it is referenced, the code will be compiled. Therefore, we must explicitly inform the compiler whether the optional command is enabled in the application configuration.

This optimization has no effect on flash size when the optional command is enabled, but it does reduce the size when the command is disabled in a specific application. In this way, we ensure there is no additional flash usage when switching to CHIP.

Reply:
The question was: if we have the ifdef _in the switch_, is that enough, without the other ifdefs, since that would let the linker remove the other (unused) code?

Reply:
It should be, the linker should be smart enough to remove the other code if it is not reference. we can also pilot "if constexpr" instead of flag